contrib/babelfishpg_common/src/collation.c

```diff
@@ -119,6 +119,40 @@ coll_translate_t coll_translations[] =
 };
 #define TOTAL_COLL_TRANSLATION_COUNT (sizeof(coll_translations)/sizeof(coll_translations[0]))
 
+/*
+ * Collation translation table.  This must be sorted by the first column (BBF collation),
+ * which contains the name of the collation to be translated from.  The second
+ * column (TSQL Collation) is the name of the collation to translate to.
+ */
+coll_translate_t reverse_coll_translations[] =
+{
+	{ "bbf_unicode_bin2", "latin1_general_bin2", 1252 },
```

Comment:
keeping a sorted table in code is not a good idea. If someone adds a wrong entry lookup can fail. Given that there are 20 odd entries only, better to do a scan rather than binary search. Add frequently used entries up front in the table as optimization

Reply:
Kept the reverse collation table unsorted and moved the frequently used entries up in the front for optimization.